pkg/storage/tpr/storage_interface_test.go

```diff
@@ -892,6 +894,114 @@ func TestDeleteWithNamespace(t *testing.T) {
 	}
 }
 
+func TestWatch(t *testing.T) {
+	const timeout = 1 * time.Second
+	keyer := getBrokerKeyer()
+	key, err := keyer.Key(request.NewContext(), name)
+	fakeCl := fake.NewRESTClient()
+	iface := getBrokerTPRStorageIFace(t, keyer, fakeCl)
+	resourceVsn := "1234"
+	predicate := storage.SelectionPredicate{}
+	obj := &sc.Broker{
+		TypeMeta:   metav1.TypeMeta{Kind: ServiceBrokerKind.String()},
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+	}
+
+	go func() {
+		if err := fakeCl.Watcher.SendObject(watch.Added, obj, 1*time.Second); err != nil {
+			t.Fatalf("error sending object %#v to watcher (%s)", *obj, err)
+		}
+		fakeCl.Watcher.Close()
```

Comment:
And note that the receive happens in the fake broker server. After that receive, the object gets encoded, sent over the "wire" (i.e. returned to the client), and then placed on the watcher channel, which is what the TPR storage interface listens on